src/js/core/UpdateStream.js

```diff
@@ -0,0 +1,100 @@
+import { Observable } from "rxjs/Observable";
+import UserSettingsStore from "#SRC/js/stores/UserSettingsStore";
+import { request } from "@dcos/http-service";
+import Util from "#SRC/js/utils/Util";
+import { SAVED_STATE_KEY } from "#SRC/js/constants/UserSettings";
+import compareVersions from "compare-versions";
+
+const CHECK_DELAY = 24 * 60 * 60 * 1000; // Once every 24 hours
+const DCOS_UI_VERSION = "dcosUIVersion";
+
+export const UpdateStreamType = Symbol("UpdateStreamType");
+
+function getFromLocalStorage(key) {
+  const value = Util.findNestedPropertyInObject(
+    UserSettingsStore.getKey(SAVED_STATE_KEY),
+    DCOS_UI_VERSION + "." + key
+  );
+
+  return value;
+}
+
+function setInLocalStorage(key, value) {
+  const savedStates = UserSettingsStore.getKey(SAVED_STATE_KEY) || {};
+  savedStates[DCOS_UI_VERSION] = {
+    dismissedVersion: getFromLocalStorage("dismissedVersion") || null,
+    lastTimeCheck: getFromLocalStorage("lastTimeCheck") || null
+  };
+  savedStates[DCOS_UI_VERSION][key] = value;
```

Comment:
Why do you think it's beneficial to store dismissed version based of `DCOS_UI_VERSION`?

Reply:
This is the global shadowing/confusing I was talking about ;)